stream_alert/classifier/payload/payload_base.py

```diff
@@ -0,0 +1,222 @@
+
+    def __new__(cls, payload_class):
+        RegisterInput._payload_classes[payload_class.service()] = payload_class
+        return payload_class
+
+    @classmethod
+    def load_for_service(cls, service, resource, raw_record):
+        """Returns the right StreamPayload subclass for this service
+
+        Args:
+            service (str): service name to load class for
+            resource (str): resource for this service
+            raw_record (str): record raw payload data
+
+        Returns:
+            StreamPayload: Loaded subclass of StreamPayload for the proper payload type
+        """
+        payload = cls._get_payload_class(service)
+        if not payload:
+            return False
+
+        return payload(resource=resource, raw_record=raw_record)
+
+    @classmethod
+    def _get_payload_class(cls, service):
+        """Returns the subclass that should handle this particular service's records
+
+        Args:
+            service (str): The service identifier for this payload
+
+        Returns:
+            StreamPayload: Subclass of StreamPayload to use for processing incoming records
+        """
+        try:
+            return cls._payload_classes[service]
+        except KeyError:
+            LOGGER.error('Requested payload service [%s] does not exist', service)
+
+
+class StreamPayload(object):
+    """StreamAlert payload object for incoming records
+
+    Attributes:
+        resource (str): The name of the resource from which this log originated.
+            Can be a kinesis stream name, SNS topic, or S3 bucket name.
+        raw_record: The record from the AWS Lambda Records dictionary.
+        log_source (str): The name of the logging application which the data
+            originated from.  This could be osquery, auditd, etc.
+        records (list): A list of parsed and typed PayloadRecord(s).
+        data_type (str): The data type of the record - json, csv, syslog, etc.
+        fully_classified (bool): Whether the payload has been successfully
+            and completely classified.
+    """
+    __metaclass__ = ABCMeta
+
+    def __init__(self, resource, raw_record):
+        """
+        Keyword Args:
+            raw_record (dict): The record to be parsed - in AWS event format
+        """
+        self.raw_record = raw_record
+        self.resource = resource
+        self.log_source = None
+        self.records = None
+        self.data_type = None
+        self.fully_classified = True
+
+    def __nonzero__(self):
+        return all([self.resource, self.data_type, self.log_source,
+                    self.records, self.fully_classified])
+
+    # For forward compatibility to Python3
+    __bool__ = __nonzero__
+
+    def __repr__(self):
+        return '<{} valid:{} log_source:{} resource:{} type:{} record:{}>'.format(
+            self.__class__.__name__,
+            bool(self),
+            self.log_source,
+            self.resource,
+            self.data_type,
+            self.records
```

Comment:
Should we also include `self.fully_classified` information here?

Reply:
this has actually significantly changed in a later PR: https://github.com/airbnb/streamalert/pull/822/files#diff-37c5fd6fd0ce64ec553cc7ac572a3173R187